src/pages/CloudModelPicker.js

```diff
@@ -145,21 +130,137 @@ class CloudModelPicker extends BaseWizardPage {
     }
   }
 
-  render() {
-    let details = '';
-    const template = this.templates.find(template => template.name === this.state.selectedModelName);
-    if(template) {
-      // details is the html help content read from model template fetched from the backend server.
-      // It should be safe to be rendered as the raw html content in the details view.
-      details = template['overview'];
+  getFilterValue = (filterName) => {
+    if (filterName === 'node-count') {
+      return this.state.filterNodeCount;
+    } else if (filterName === 'hypervisor-type') {
+      return this.state.filterHypervisorType;
+    } else if (filterName === 'storage-type') {
+      return this.state.filterStorageType;
+    } else if (filterName === 'network-type') {
+      return this.state.filterNetworkType;
     }
+  }
 
-    const btns = this.simpleModels.map((name,idx) =>
+  handleFilter = (filterName, value) => {
+    if (filterName === 'node-count') {
+      this.setState({filterNodeCount: value});
+    } else if (filterName === 'hypervisor-type') {
+      this.setState({filterHypervisorType: value});
+    } else if (filterName === 'storage-type') {
+      this.setState({filterStorageType: value});
+    } else if (filterName === 'network-type') {
+      this.setState({filterNetworkType: value});
+    }
+  }
+
+  filterTemplates = () => {
+    return this.templates.filter((template) => {
+      return (this.state.filterNodeCount === 'none' ||
+          template.metadata.nodeCount === this.state.filterNodeCount) &&
+        (this.state.filterHypervisorType === 'none' ||
+          template.metadata.hypervisor.includes(this.state.filterHypervisorType)) &&
+        (this.state.filterStorageType === 'none' ||
+          template.metadata.storage === this.state.filterStorageType) &&
+        (this.state.filterNetworkType === 'none' ||
+          template.metadata.network === this.state.filterNetworkType);
+    });
+  }
+
+  // util to find unique values from array of arrays
+  getUniqueValues(inputArray) {
+    let uniques = [];
+    inputArray.forEach((input) => {
+      input.forEach((inp) => {
+        if (!uniques.includes(inp)) {
+          uniques.push(inp);
+        }
+      });
+    });
+    return uniques;
+  }
+
+  processFilters = (templates) => {
+    // 'newFilters' keeps track of ALL filter options which changes every time the user makes a new
+    // selection by looking at metadata of all templates and sorting out the available values by
+    // filter type then displaying those values as filter options
+    let newFilters = JSON.parse(JSON.stringify(FILTERS));
```

Comment:
This can't be right. It appears that this converting to a string and back again (YUCK) just to make a copy so that it can be modified in place. It would be more idiomatic to use FILTERS.map, which iterates through and returns a new array without touching the original